pkg/module/sync/runner_impl.go

```diff
@@ -121,22 +121,33 @@ func (r *RunnerImpl) updateModuleStatusFromExistingModules(
 
 	for idx := range modules {
 		module := modules[idx]
-		manifestAPIVersion, manifestKind := module.Object.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
+		manifestObject := module.Object.(*v1beta1.Manifest)
+		manifestAPIVersion, manifestKind := manifestObject.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
 		templateAPIVersion, templateKind := module.Template.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
+		moduleCRAPIVersion := ""
+		moduleCRKind := ""
+		if manifestObject.Spec.Resource != nil {
```

Comment:
If the `manifestObject.Spec.Resource` is nil, then the ModuleStatus.Resource should also nil, you don't need to filling some empty object.

Reply:
that's why design the `Resource *TrackingObject` as a pointer reference, so if it's not initiated, this filed will be empty.